### ISSUE TITLE
Mirror in-game role order

### ISSUE THREAD

**Issue by psikoi** [wise-old-man/wise-old-man#1490]
If we add a new model to our database that looks something like this:

```typescript
// GroupRoleOrder
{
    groupId: number;
    role: GroupRole;
    index: number;
}
```

We could then populate this information via de "edit group" endpoint (which is used by the RL plugin's "Sync" feature):

`PUT /groups/:id`

```
{
    name: ...
    clanChat: ...
    roleOrder: [
        { role: "magician", index: 0 },
        { role: "legend", index: 1 },
        { role: "owner", index: 2 },
   ]
}
```

and then fetch this information, and return a sorted members list on `GET /groups/:id`

I don't think this would require any frontend work. Unless we decide this should also be configurable in the website, but for now, it seems fine to be a RL Sync feature only.

**Comment by aleccaputo** [comment]
What does the migration for this look like? Is there no migration and only net-new when someone syncs the plugin?

**Comment by psikoi** [comment]
> What does the migration for this look like? Is there no migration and only net-new when someone syncs the plugin?

Yea, I think so

**Comment by psikoi** [comment]
Implemented in #1496 (still needs RL plugin integration)
